include/WaveParallel.hpp

```diff
@@ -0,0 +1,338 @@
+     * on whether the triangulation is built from a file or not.
+     */
+    void complete_setup();
+
+    /**
+     * @brief Constructs the mass matrix and laplace matrix for the problem. 
+     * 
+     */
+    void assemble_matrices();
+
+    /**
+     * @brief Runs the solver by iteratively computing the right hand side of the position equation
+     * (u), solving the first system of equations (u_n+1) with the conjugate gradient method,
+     * computing the right hand side of the velocity equation with the newly produced u_n+1,
+     * and solving it by applying the conjugate gradient method. 
+     */
+    void run();
+
+    void print_timer_data() const;
+
+    WaveEquationParallel(
+        const unsigned int& degree_,
+        const double& interval_,
+        const double& time_step_, 
+        const double& theta_
+    )
+    :   degree (degree_)
+    ,   interval (interval_)
+    ,   time_step (time_step_)
+    ,   theta (theta_)
+    ,   mpi_size(Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD))
+    ,   mpi_rank(Utilities::MPI::this_mpi_process(MPI_COMM_WORLD))
+    ,   triangulation(MPI_COMM_WORLD)
+    ,   pcout(std::cout, mpi_rank == 0)
+    ,   timer(pcout, TimerOutput::summary, TimerOutput::wall_times)
+    {}
+
+    /**
+     * @brief Class describing the initialization value of the height of each point in the
+     * computational domain.
+     * 
+     * @tparam dim 
+     */
+    class InitialU : public Function<dim>
+    {
+    public:
+        InitialU(){}
+
+        virtual double value(const Point<dim>& /*p*/, const unsigned int /*component*/ = 0) const override
+        {
+            return 0.0;
+        }
+
+    };
+
+    /**
+     * @brief Class describing the initialization value of the velocity of each point in the computational
+     * domain.
+     * 
+     * @tparam dim number of physical dimensions of the problem
+     */
+    class InitialV : public Function<dim>
+    {
+    public:
+        InitialV(){}
+
+        virtual double value(const Point<dim>& /*p*/, const unsigned int /*component*/ = 0) const override
+        {
+            return 0.0;
+        }
+
+    };
+
+    /**
+     * @brief Class describing the boundary values related to position. 
+     * 
+     */
+    class BoundaryU : public Function<dim>
+    {
+    public:
+        BoundaryU(){}
+
+        virtual double value(const Point<dim>& /*p*/, const unsigned int /*component*/ = 0) const override
+        {   
+            return 0.0;
+        }
+
+    };
+
+    /**
+     * @brief Class describing the boundary values related to velocity. They can be extracted by computing
+     * the derivative of the function describin the boundary values for position.
+     * 
+     */
+    class BoundaryV : public Function<dim>
+    {
+    public:
+        BoundaryV(){}
+
+        virtual double value(const Point<dim>& /*p*/, const unsigned int /*component*/ = 0) const override
+        {
+            return 0.0;
+        }
+
+    };
```

Comment:
The problem does not involve any boundary conditions on the velocity auxiliary variable (since it inherits boundary conditions from the displacement variable). Are you sure that this class is necessary?